drivers/volume/csi/csi.go

```diff
@@ -1370,7 +1372,7 @@ func (c *csi) restoreVolumeSnapshot(
 		vsObj.Spec.Source.PersistentVolumeClaimName = nil
 		vsObj.Spec.Source.VolumeSnapshotContentName = &vsc.(*kSnapshotv1.VolumeSnapshotContent).Name
 		vsObj.Namespace = namespace
-		vs, err := c.snapshotClient.SnapshotV1().VolumeSnapshots(namespace).Create(context.TODO(), vsObj, metav1.CreateOptions{})
```

Comment:
this can remain as "vs" but if it already exists then do a get and return the updated vs object.

Reply:
Yes, no need introduce new variable. In the case of already exists, get the lates one from the setup and return it

Reply:
done.